Assets/CarriageManager.cs

```diff
@@ -0,0 +1,143 @@
+
+    private List<GameObject> carriagesToSpawn;
+    [SerializeField] private List<GameObject> spawnedCarriages;
+    [SerializeField] private CarriageData currentCarriage, oldCarriage;
+
+    //SPAWNING
+    [SerializeField] private Transform firstSpawn;
+    [SerializeField] private Transform nextSpawnPos,prevSpawnPos;
+
+    //MOVING
+    private void Awake()
+    {
+    }
+    private void Start()
+    {
+        InitialiseCarriages();
+    }
+
+    private void InitialiseCarriages()
+    {
+        //ALGORITHM CAN BE ALTERED TO BE RANDOMISED ON EVERY NEW PLAYTHROUGH OR KEPT AS IS
+
+        carriagesToSpawn = new List<GameObject>(carriagePrefabs.Count);
+        foreach (GameObject _carriage in carriagePrefabs)
+            carriagesToSpawn.Add(_carriage);
+
+
+        //PREPARE SPAWN FIRST CARRIAGE
+        nextSpawnPos = firstSpawn;
+
+        //NAMING
+        int _count = 0;
+
+        //SPAWN SUBSEQUENT CARRIAGES
+        foreach (GameObject _carriage in carriagesToSpawn)
+        {
+            
+            GameObject _spawned = Instantiate(_carriage, nextSpawnPos.position, nextSpawnPos.rotation);
+
+            //NAMING
+            _count++;
+            _spawned.name = "Carriage " + (_count);
+            _spawned.GetComponent<CarriageData>().InitialiseId(_count);
+
+            spawnedCarriages.Add(_spawned);
+
+            //UPDATE NEXT SPAWN
+            nextSpawnPos = _spawned.GetComponent<CarriageData>().exitPos;
+            
+        }
+
+        //INITIALISE CURRENT/OLD CARRIAGE SETTINGS
+        currentCarriage = spawnedCarriages[2].GetComponent<CarriageData>();
+        oldCarriage = currentCarriage;
+        prevSpawnPos = spawnedCarriages[0].gameObject.transform;
+
+    }
+    public void Update_currentCarriage(CarriageData _current)
+    {
+        currentCarriage = _current;
+        Debug.Log("current: " + currentCarriage.carriageId + ". old: " + oldCarriage.carriageId);
+        
+        
+        //MOVE CARRIAGE FROM BACK TO FRONT
+        //Whether new carriage is infront of old carriage, or new carriage is the highest value of the carriageID list possible.
+        if ((currentCarriage.carriageId > oldCarriage.carriageId) && currentCarriage.carriageId != oldCarriage.carriageId)
+        {
+           
+            GameObject _backCarriage = spawnedCarriages[0];
+
+            _backCarriage.transform.position = nextSpawnPos.position;
+            _backCarriage.transform.rotation = nextSpawnPos.rotation;
+            
+            spawnedCarriages.Add(_backCarriage.gameObject);
+            _backCarriage.GetComponent<CarriageData>().carriageId+=spawnedCarriages.Count;
+            nextSpawnPos = spawnedCarriages[spawnedCarriages.Count - 1].GetComponent<CarriageData>().exitPos;
+            
+            //NEW BACK CARRIAGE
+            spawnedCarriages.RemoveAt(0);
+            prevSpawnPos = spawnedCarriages[0].gameObject.transform;
+
+
+            
+            Debug.Log("attempt move");
```

Comment:
remove, this will spam the debug log

Reply:
Will remove in future branches. Was used in testing.